src/compiler/types.ts

```diff
@@ -2813,6 +2814,7 @@ namespace ts {
 
     export interface CompilerHost extends ModuleResolutionHost {
         getSourceFile(fileName: string, languageVersion: ScriptTarget, onError?: (message: string) => void): SourceFile;
+        getSourceFileByPath?(fileName: string, path: Path, languageVersion: ScriptTarget, onError?: (message: string) => void): SourceFile;

```

Comment:
we need both on the LS since file name is used to query managed side for the script snapshot. With this order it is just consistent with `getSourceFile` where `fileName` is a first argument